packages/frontend/src/pages/CreateService.svelte

```diff
@@ -8,63 +8,61 @@ import { onMount } from 'svelte';
-export let trackingId: string | undefined = undefined;
+interface Props {
+  // The tracking id is a unique identifier provided by the
+  // backend when calling requestCreateInferenceServer
+  trackingId?: string;
+}
+
+let { trackingId }: Props = $props();
 
 // List of the models available locally
-let localModels: ModelInfo[];
-$: localModels = $modelsInfo.filter(model => model.file);
+let localModels: ModelInfo[] = $derived($modelsInfo.filter(model => model.file));
 
 // The container provider connection to use
-let containerProviderConnection: ContainerProviderConnectionInfo | undefined = undefined;
+let containerProviderConnection: ContainerProviderConnectionInfo | undefined = $state(undefined);
 
 // Filtered connections (started)
-let startedContainerProviderConnectionInfo: ContainerProviderConnectionInfo[] = [];
-$: startedContainerProviderConnectionInfo = $containerProviderConnections.filter(
-  connection => connection.status === 'started',
+let startedContainerProviderConnectionInfo: ContainerProviderConnectionInfo[] = $derived(
+  $containerProviderConnections.filter(connection => connection.status === 'started'),
 );
 
-// Select default connection
-$: if (containerProviderConnection === undefined && startedContainerProviderConnectionInfo.length > 0) {
-  containerProviderConnection = startedContainerProviderConnectionInfo[0];
-}
-
 // The containerPort is the bind value to form input
-let containerPort: number | undefined = undefined;
+let containerPort: number | undefined = $state(undefined);
 // The model is the bind value to ModelSelect form
-let model: ModelInfo | undefined = undefined;
+let model: ModelInfo | undefined = $state(undefined);
 // If the creation of a new inference service fail
-let errorMsg: string | undefined = undefined;
-// The trackedTasks are the tasks linked to the trackingId
-let trackedTasks: Task[];
-
-// has an error been raised
-let error: boolean = false;
-
+let errorMsg: string | undefined = $state(undefined);
 // The containerId will be included in the tasks when the creation
 // process will be completed
-let containerId: string | undefined = undefined;
-$: available = containerId && $inferenceServers.some(server => server.container.containerId);
-
-$: loading = trackingId !== undefined && !error;
-
-$: {
+let containerId: string | undefined = $state(undefined);
+// available means the server is started
+let available: boolean = $derived(!!containerId && $inferenceServers.some(server => server.container.containerId));
+// loading state
+let loading = $derived(trackingId !== undefined && !errorMsg);
+
+// Select default model
+$effect(() => {
   if (!model && localModels.length > 0) {
     model = localModels[0];
   }
-}
+});
+
+// Select default connection
+$effect(() => {
+  if (containerProviderConnection === undefined && startedContainerProviderConnectionInfo.length > 0) {
+    containerProviderConnection = startedContainerProviderConnectionInfo[0];
+  }
+});
```

Comment:
do we need 2 effect methods ?

Reply:
No, actually there is no need to have multiple effect. Combined!